src/plugins/config/rabbitmq_config.py

```diff
@@ -0,0 +1,29 @@
+"""
+RabbitMQ Configuration
+
+The connection with RabbitMQ is setup and initialised using the information in this module
+
+This includes Exchange, Queue and Routing key information that is specific for GOB
+"""
+import os
+import pika
+
+MESSAGE_BROKER = os.getenv("MESSAGE_BROKER_ADDRESS", "localhost")
+MESSAGE_BROKER_PORT = os.getenv("MESSAGE_BROKER_PORT", 15672)
+MESSAGE_BROKER_VHOST = os.getenv("MESSAGE_BROKER_VHOST", "gob")
+MESSAGE_BROKER_USER = os.getenv("MESSAGE_BROKER_USERNAME", "guest")
+MESSAGE_BROKER_PASSWORD = os.getenv("MESSAGE_BROKER_PASSWORD", "guest")
+
+CONNECTION_PARAMS = pika.ConnectionParameters(
+    host=MESSAGE_BROKER,
+    virtual_host=MESSAGE_BROKER_VHOST,
+    credentials=pika.PlainCredentials(username=MESSAGE_BROKER_USER,
+                                      password=MESSAGE_BROKER_PASSWORD),
+    heartbeat_interval=1200,
+    blocked_connection_timeout=600
+)
+
+EXCHANGE = "gob.workflow"
+REQUEST_KEY = "workflow.request"
```

Comment:
Why not to send messages directly to the services actually? Using workflow queue means we still need workflow for airflow to function...

We can mark messages as "airflow" message, services will keep this mark intact. Workflow will recognize "workflow" messages. Airflow can do the same by recognizing "airflow" messages in the generic result queue. This way workflow and airflow can function in parallel but airflow will not be dependent on workflow.

Reply:
GOB Workflow management

GOB workflows can be triggered from within GOB (using GOB Workflow start) or from outside GOB (Jenkins, Airflow, Camunda).

The following needs to be kept in mind:

- There will never be one workflow tool that uniquely controls GOB. It should be possible to control GOB as you like.

- All jobs that run within GOB should be known by GOB. This is essential as it can be used to eliminate duplicate executions. Although no damage is done (GOB is idempotent) it just wastes energy and should be prevented.

- External workflow management tools should require minimal knowledge of the internal GOB architecture. They should never know something about individual GOB components. GOB should always be maximally free to change it’s components, queues, routing keys, database schemes or whatsoever. The more the outside knows, the harder it is to maintain GOB.

When workflows are triggered by remote tools this should always go via GOB Workflow. GOB Workflow is the only component that is known to the outside. In order to access GOB Workflow it may be required to know the host and name of the docker (as it is in Jenkins, or when workflows are manually triggered) or the exchange, queue and request key (as for Airflow).

A special start package has been made to allow for programmatic triggers. A hook in the message handling has been made to allow for message triggers.

As every workflow and workflow step passes via GOB Workflow, GOB will always know which workflows and steps are active. The GOB Management interface (Iris) will always show the status of GOB, independent of which workflow tool triggered its execution.

It is an essential requirement that whatever tool is used, Iris will always be able to show the status of any GOB workflow that runs or has run.

This is the reason that no workflow tool should ever send messages directly to individual GOB modules. Too much knowledge is required about message formats, queues or whatever more (which makes GOB less free to change and to be maintained).

By bypassing GOB Workflow GOB has no knowledge of its active processes. Just use the hook in the message header and any tool can manage GOB with minimal knowledge and keeps GOB maximally free to grow, change and do whatever is required to keep it working.

Reply:
Thanks. I understood the concept.